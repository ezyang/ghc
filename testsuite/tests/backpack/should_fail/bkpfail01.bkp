unit p where
    signature H where
        data H = H
    module A where
        import H
        data A = A H

unit q where
    module H where
        data S = S

unit r where
    include q
    include p
    module B where
        import A
        x = A H
