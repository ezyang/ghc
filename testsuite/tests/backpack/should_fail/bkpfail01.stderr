[1 of 3] Processing p
  [1 of 3] Compiling H[boot]          ( p/H.hs-boot, nothing )
  [2 of 3] Compiling H[merge]         ( p/H.hs-boot-merge, nothing )
  [3 of 3] Compiling A                ( p/A.hs, nothing )
[2 of 3] Processing q
  Instantiating main-q
  [1 of 1] Compiling H                ( q/H.hs, nothing )
[3 of 3] Processing r
  Instantiating main-r
  [1 of 2] Including main-q
  [2 of 2] Including main-p
    Instantiating main-p(main-q:H)
    [1 of 3] Compiling H[boot]          ( p/H.hs-boot, nothing )

bkpfail01.bkp:2:5: error:
    Not in scope: type constructor or class ‘H’

bkpfail01.bkp:2:5: error: Not in scope: data constructor ‘H’
